guitar-midi/src/main.cpp

```diff
@@ -100,13 +108,17 @@ void setup()
 
 void loop()
 {
-  updateButton(PIN_GREEN, 48);
-  updateButton(PIN_RED, 49);
-  updateButton(PIN_YELLOW, 50);
-  updateButton(PIN_BLUE, 51);
-  updateButton(PIN_ORANGE, 52);
-  updateButton(PIN_UP, 59);
-  updateButton(PIN_DOWN, 58);
+  auto now = millis();
+  updateButton(PIN_GREEN,  48, now, pinState[0], debounceState[0]);
+  updateButton(PIN_RED,    49, now, pinState[1], debounceState[1]);
+  updateButton(PIN_YELLOW, 50, now, pinState[2], debounceState[2]);
+  updateButton(PIN_BLUE,   51, now, pinState[3], debounceState[3]);
+  updateButton(PIN_ORANGE, 52, now, pinState[4], debounceState[4]);
+  // Pins UP and down share the same debounce state intentionally.
+  // When strumming with a particular pattern, the button flies back and opens 
+  // the other switch, which is not something we want.
+  updateButton(PIN_UP,     59, now, pinState[5], debounceState[5]); 
+  updateButton(PIN_DOWN,   58, now, pinState[6], debounceState[5]);
```

Comment:
Another way to do the same without passing everything around (not sure if it's better or worse... probably much worse, but the copy part is veeeery explicit) 😄

```c
  now = millis(); // global
  updateButton(PIN_GREEN, 48);
  // ...
  updateButton(PIN_ORANGE, 52);
  // Pins UP and DOWN share the same debounce state intentionally.
  // When strumming with a particular pattern, the button flies back and opens 
  // the other switch, which is not something we want.
  updateButton(PIN_UP, 59); 
  debounceState[PIN_DOWN] = debounceState[PIN_UP]; 
  updateButton(PIN_DOWN, 58);
  debounceState[PIN_UP] = debounceState[PIN_DOWN]; 
```

Reply:
I think it's nicer to explicitly pass in state. That said, I modified this to be more explicit, see if it looks better.

Reply:
nice